enterprise/server/test/integration/remote_execution/remote_execution_test.go

```diff
@@ -1759,7 +1759,98 @@ func TestAppShutdownDuringExecution(t *testing.T) {
 
 	// Simulate the shutdown grace period elapsing, which should cause a
 	// hard stop of the gRPC server on app 1.
-	time.Sleep(100 * time.Millisecond)
+	time.Sleep(200 * time.Millisecond)
+	app1.Stop()
+
+	eg := &errgroup.Group{}
+	for _, cmd := range cmds {
+		cmd := cmd
+		eg.Go(func() error {
+			// Maybe let the command continue execution for a bit, then exit.
+			randSleepMillis(0, 50)
+
+			cmd.Exit(0)
+			res := cmd.Wait()
+			require.Equal(t, 0, res.ExitCode)
+			return res.Err
+		})
+	}
+	err := eg.Wait()
+	require.NoError(t, err)
+
+	// Make sure we only ever started a single task execution per command (tasks
+	// should not be re-executed just because the app goes down).
+	tasksStartedCount := testmetrics.CounterValue(t, metrics.RemoteExecutionTasksStartedCount) - initialTasksStartedCount
+	require.Equal(t, float64(len(cmds)), tasksStartedCount, "no tasks should have been retried")
+}
```

Comment:
this stuff was not actually added, it's just identical to the last few lines of code in `TestAppShutdownDuringExecution_LeaseTaskRetried`